sklearn/preprocessing/_polynomial.py

```diff
@@ -765,8 +784,8 @@ def _get_base_knot_positions(X, n_knots=10, knots="uniform", sample_weight=None)
             # `else` is therefore safe.
             # Disregard observations with zero weight.
             mask = slice(None, None, 1) if sample_weight is None else sample_weight > 0
-            x_min = np.amin(X[mask], axis=0)
-            x_max = np.amax(X[mask], axis=0)
```

Comment:
Similarly, if `np.isnan(x_min)` (which means we have an all-nan column), then replacing `x_min` and `x_max` by 0 might be helpful to treat this column as a regular constant column and always encode the output to 0.0, whatever the input.